sttz.InstallUnity/Installer/Platforms/WIndowsPlatform.cs

```diff
@@ -0,0 +1,316 @@
+                Logger.LogDebug($"Found version {versionInfo.ProductVersion}");
+                unityInstallations.Add(new Installation {
+                    executable = modulesJsonPath,
+                    path = unityCandidate,
+                    version = new UnityVersion(versionInfo.ProductVersion.Substring(0, versionInfo.ProductVersion.LastIndexOf(splitCharacter)))
+                });
+            }
+            return unityInstallations;
+        }
+
+        public async Task Install(UnityInstaller.Queue queue, UnityInstaller.QueueItem item, CancellationToken cancellation = default)
+        {
+            if (item.package.name != PackageMetadata.EDITOR_PACKAGE_NAME && !installedEditor)
+            {
+                throw new InvalidOperationException("Cannot install package without installing editor first.");
+            }
+
+            var installPath = GetInstallationPath(installing.version, installationPaths);
+            // TODO: start info runas
+            var result = await RunAsAdmin(item.filePath, $"/S /D={installPath}");
+            if (result.exitCode != 0)
+            {
+                throw new Exception($"Failed to install {item.filePath} output: {result.output} / {result.error}");
+            }
+
+            if (item.package.name == PackageMetadata.EDITOR_PACKAGE_NAME)
+            {
+                installedEditor = true;
+            }
+        }
+
+        public async Task MoveInstallation(Installation installation, string newPath, CancellationToken cancellation = default)
+        {
+            // do nothing
+        }
+
+        public async Task PrepareInstall(UnityInstaller.Queue queue, string installationPaths, CancellationToken cancellation = default)
+        {
+            if (installing.version.IsValid)
+                throw new InvalidOperationException($"Already installing another version: {installing.version}");
+
+            installing = queue.metadata;
+            this.installationPaths = installationPaths;
+            installedEditor = false;
+
+            // Check for upgrading installation
+            if (!queue.items.Any(i => i.package.name == PackageMetadata.EDITOR_PACKAGE_NAME))
+            {
+                var installs = await FindInstallations(cancellation);
+                var existingInstall = installs.Where(i => i.version == queue.metadata.version).FirstOrDefault();
+                if (existingInstall == null)
+                {
+                    throw new InvalidOperationException($"Not installing editor but version {queue.metadata.version} not already installed.");
+                }
+
+                installedEditor = true;
+            }
+        }
+
+        public async Task<bool> PromptForPasswordIfNecessary(CancellationToken cancellation = default)
+        {
+            // Don't care about password. The system will ask for elevated priviliges automatically
+            return true;
```

Comment:
Guessing this has a warning about async wihtout awaits. Return Task.FromResult(true) instead